packages/graphql_client/lib/src/farms.dart

```diff
@@ -1,43 +1,45 @@
 import 'package:graphql_client/graphql_client.dart';
 import 'package:graphql_client/models/farms.dart';
+import 'package:graphql_client/models/public_ips.dart';
 
 class TFFarms {
   final GraphQLClient gqlClient;
 
   TFFarms(this.gqlClient);
 
-  Future<List<FarmInfo>> listFarmsByTwinId(
-      ListFarmsByTwinIdOptions options) async {
-    print(options.twinId);
+  Future<List<FarmInfo>> listFarms(FarmsQueryOptions? queryOptions,
+      FarmsReturnOptions? returnOptions) async {
+    final queryString = queryOptions?.toString() ?? "";
+    final returnString = returnOptions?.toString() ?? "id \n";
+
     final body = '''
-query getFarms(\$twinId: Int!) {
-  farms(where: {twinID_eq: \$twinId}) {
-    farmID
-    name
-    certification
-    dedicatedFarm
-    pricingPolicyID
-    stellarAddress
-    twinID
-  }
-}''';
+            query Farms{
+              farms $queryString {
+                $returnString
+              }
+            }''';
+
+    print(body);
 
     try {
-      final response =
-          await gqlClient.query(body, variables: {'twinId': options.twinId});
+      final response = await gqlClient.query(body);
+      print(response['data']);
       List<FarmInfo> farms =
           (response['data']['farms'] as List<dynamic>).map((farmsData) {
         return FarmInfo(
+          id: farmsData['id'],
           farmID: farmsData['farmID'] ?? 0,
           certification: farmsData['certification'] ?? '',
           dedicatedFarm: farmsData['dedicatedFarm'] ?? false,
           name: farmsData['name'] ?? '',
           pricingPolicyID: farmsData['pricingPolicyID'] ?? 0,
           stellarAddress: farmsData['stellarAddress'] ?? '',
           twinID: farmsData['twinID'] ?? 0,
+          publicIps: farmsData['publicIps'] != null
+              ? PublicIpsInfo.fromJson(farmsData['publicIps'])
+              : null,
         );
       }).toList();
-
       return farms;
     } catch (err) {
       throw err;
```

Comment:
- what is the point of the catch here, the error already will be propagated from query or the block of code after it will fail
- either u handle the crash or leave it.